### ISSUE TITLE
Error in sorting and groups disappear with plugins assigned to groups + Load After rules

### ISSUE THREAD

**Issue by RealAntithesis** [loot/loot#1188]
Game: Skyrim Special Edition
Version: 0.15.1 (build 82c756e1)

Issue: Groups flowchart editor is blank after adding plugins to (user-defined) groups and then also adding Load After rules for those plugins. This also resulted in an 'general' error that the plugins could not be sorted when attempting to sort.

Resolved: since I just wanted to use the Load After data rather then the groups, but I couldn't edit the groups any longer in the app, I opened up the userlist.yaml file and deleted those entries for the user-defined groups I had set up earlier including the group data for the plugins. This fixed the error and allowed sorting to work again (and the groups flowchart showed up again in the Groups Editor).

I guess there was a circular dependency between the groups and Load After orders, but this shouldn't have resulted in the groups disappearing and a general error popping up when attempting to sort that didn't point to the actual cause.

**Comment by CptMcSplody** [comment]
@RealAntithesis There were three groups removed in https://github.com/loot/skyrimse/commit/d0fcf8e8f4c39f19373b5c196841316c33a923b1. They had become redundant with the other changes.

> Groups flowchart editor is blank after adding plugins to (user-defined) groups

Yeah I also had the same.
![image](https://user-images.githubusercontent.com/1944639/76247805-10eed680-6238-11ea-9f72-e695b76aa660.png)

> Uncaught Error: Can not create edge `TestUnderrides` with nonexistant source `Underrides`

That's the error I receive for the below user rule.
![image](https://user-images.githubusercontent.com/1944639/76247104-e5b7b780-6236-11ea-9648-25cecb66b53e.png)

If I add Underrides to the userlist the error also goes away.

```yaml
groups:
  - name: 'Test'
    after:
      - Underrides
  - name: 'Underrides'
  - name: 'default'
```

So It could be prevented If a group was copied from the masterlist while it was being referenced in the userlist.

**Comment by Ortham** [comment]
I can't replicate this, can you share steps for getting the error from scratch?

**Comment by CptMcSplody** [comment]
Change the settings to use 14.0 branch & update the masterlist.
Open the group editor and add a new group `Test` that loads before or after the `Underrides` group.
Change the settings to use 15.0 branch & update the masterlist.
Open the group editor again, it should now be blank.

**Comment by Ortham** [comment]
Hmm, the editor doesn't turn blank for me if I set `Test` to load before `Underrides`, I get:

![Screenshot (76)](https://user-images.githubusercontent.com/2430738/76350640-069a0e80-6304-11ea-8d4c-bdc1dc2fe0e7.png)

I.e. the `Underrides` group has turned into a userlist group.

However, I do get the blank editor if `Test` loads after `Underrides`.

I think the correct behaviour in this case would be to automatically turn missing groups into userlist groups. Do you agree?

**Comment by CptMcSplody** [comment]
> I think the correct behaviour in this case would be to automatically turn missing groups into userlist groups. Do you agree?

Yeah I think that would fine.

**Comment by Ortham** [comment]
OK, let me know if [this build](https://bintray.com/loot/snapshots/download_file?file_path=loot_0.15.1-5-gd565efb_add-missing-groups.7z) works how you expect it should.

**Comment by CptMcSplody** [comment]
> OK, let me know if [this build](https://bintray.com/loot/snapshots/download_file?file_path=loot_0.15.1-5-gd565efb_add-missing-groups.7z) works how you expect it should.

Yes it's working as expected.

**Comment by Ortham** [comment]
OK, merged in 7173b96663ec64a4524c18f7f5b58ea5dd851062.
